src/pocketmine/level/Position.php

```diff
@@ -89,6 +89,62 @@ public function setLevel(Level $level = null){
 		return $this;
 	}
 
+	/**
+	 * @param Position|int $x
```

Comment:
This is backwards-incompatible, disallowing passing a `Vector3` to `Position->add()`.
This also violates covariance, because this method is expected to work but conditionally doesn't:
```php
function foo(Vector3 $a) : Vector3 {
    return $a->add(new Vector3(1, 2, 3));
}
```
Calling `foo` with a `Position` object leads to a crash (trying to add `$this->x` to a `Vector` object), while `foo` did not even need to depend on pmmp.